packages/swap/src/helpers/orders/makeOrderCalculations.ts

```diff
@@ -0,0 +1,182 @@
+import {Quantities} from '../../utils/quantities'
+
+export const makeOrderCalculations = ({
+  orderType,
+  amounts,
+  limitPrice,
+  slippage,
+  pools,
+  primaryTokenId,
+  lpTokenHeld,
+  action,
+}: Readonly<{
+  orderType: Swap.OrderType
+  amounts: {
+    sell: Balance.Amount
+    buy: Balance.Amount
+  }
+  limitPrice?: Balance.Quantity
+  pools: ReadonlyArray<Swap.Pool>
+  lpTokenHeld?: Balance.Amount
+  slippage: number
+  primaryTokenId: Balance.TokenInfo['id']
+  action?: 'buy' | 'sell'
+}>): Array<SwapOrderCalculation> => {
+  const isLimit = orderType === 'limit'
+  const maybeLimitPrice = isLimit ? limitPrice : undefined
+
+  return pools.map<SwapOrderCalculation>((pool) => {
+    const buy =
+      action === 'sell'
+        ? getBuyAmount(pool, amounts.sell, maybeLimitPrice)
+        : amounts.buy
+    const sell =
+      action === 'buy'
+        ? getSellAmount(pool, amounts.buy, maybeLimitPrice)
+        : amounts.sell
+
+    const marketPrice = getMarketPrice(pool, sell)
+    // recalculate price base, limit is user's input, market from pool
+    const priceBase = isLimit ? limitPrice ?? marketPrice : marketPrice
+
+    // calculate buy quantity with slippage
+    const buyAmountWithSlippage: Balance.Amount = {
+      quantity: getQuantityWithSlippage(buy.quantity, slippage),
+      tokenId: buy.tokenId,
+    }
+
+    // pools that with not enough supply will be filtered out
+    const isBuyTokenA = buy.tokenId === pool.tokenA.tokenId
+    const poolSupply = isBuyTokenA ? pool.tokenA.quantity : pool.tokenB.quantity
+    const hasSupply = !Quantities.isGreaterThan(
+      buy.quantity,
+      poolSupply ?? Quantities.zero,
+    )
+
+    // lf is sell side % of quantity ie. XToken 100 * 1% = 1 XToken
+    const liquidityFee: Balance.Amount = getLiquidityProviderFee(pool.fee, sell)
+
+    const [ptPriceBuy, ptPriceSell] = isBuyTokenA
+      ? [pool.ptPriceTokenA, pool.ptPriceTokenB]
+      : [pool.ptPriceTokenB, pool.ptPriceTokenA]
+
+    // ffee is based on PT value range + LP holding range (sides may need conversion, when none is PT)
+    const frontendFeeInfo = getFrontendFee({
+      sell,
+      buy,
+      lpTokenHeld,
+      primaryTokenId,
+      sellInPrimaryTokenValue: {
+        tokenId: primaryTokenId,
+        quantity: asQuantity(
+          new BigNumber(amounts.sell.quantity)
+            .times(ptPriceSell)
+            .integerValue(BigNumber.ROUND_CEIL),
+        ),
+      },
+      buyInPrimaryTokenValue: {
+        tokenId: primaryTokenId,
+        quantity: asQuantity(
+          new BigNumber(amounts.buy.quantity)
+            .times(ptPriceBuy)
+            .integerValue(BigNumber.ROUND_CEIL),
+        ),
+      },
+    })
+
+    // transform fees in terms of sell side quantity * pt price (unit of fees)
+    // it applies market price always
+    const feeInSellSideQuantities = {
+      batcherFee: new BigNumber(pool.batcherFee.quantity)
+        .times(ptPriceSell)
```

Comment:
If price ada/tokeX you need to calc ( batcher_fee / tokenX_price ) instead ( batcher_fee * tokenX_price )